File: ios/Models/Anchor.swift
import Foundation

struct Anchor: Codable, Identifiable {
    enum AnchorType: String, Codable {
        case pin
        case layup
        case reticle
        case groundPlane
    }

    let id: UUID
    let sessionID: UUID
    let type: AnchorType
    let position: SIMD3<Double>
    let normal: SIMD3<Double>
    let stabilityConfidence: Double
    let lastRevalidatedAt: Date
    var driftMeters: Double
}

File: android/app/src/main/java/com/golfiq/hud/model/Anchor.kt
package com.golfiq.hud.model

import java.time.Instant
import java.util.UUID

data class Anchor(
    val id: UUID,
    val sessionId: UUID,
    val type: AnchorType,
    val position: Triple<Double, Double, Double>,
    val normal: Triple<Double, Double, Double>,
    val stabilityConfidence: Double,
    val lastRevalidatedAt: Instant,
    val driftMeters: Double,
) {
    enum class AnchorType { PIN, LAYUP, RETICLE, GROUND_PLANE }
}